ocean_provider/validation/provider_requests.py

```diff
@@ -128,6 +128,39 @@ def validate_download_signature(self, value, params, **kwargs):
 
         return False
 
+    def validate_decrypt_signature(self, value, params, **kwargs):
+        """
+        Validates a signature using the decrypterAddress.
+
+        parameters:
+          - name: value
+            type: string
+            description: Value of the field being validated
+          - name: params
+            type: list
+            description: The list of parameters defined for the rule,
+                         i.e. names of other fields inside the request.
+        """
+        self._assert_params_size(size=4, params=params, rule="signature")
+        transaction_id = self._attribute_value(params[0])
+        data_nft_address = self._attribute_value(params[1])
+        decrypter_address = self._attribute_value(params[2])
+        chain_id = self._attribute_value(params[3])
+        nonce = self._attribute_value(params[4])
+
+        if transaction_id:
+            original_msg = f"{transaction_id}{decrypter_address}{chain_id}"
```

Comment:
please only keep what is different between the ifs e.g.
`first_arg = transaction_id if transaction_id else data_nft_address`
`original_msg = f"{first_arg}{decrypter_address}{chain_id}"`

Reply:
Done. Thanks for the comment!